examples/daemonsetsharding/deployment-no-node-pods.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: exporter
+    app.kubernetes.io/name: kube-state-metrics-global
+    app.kubernetes.io/version: 2.10.0
+  name: kube-state-metrics-global
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kube-state-metrics
```

Comment:
nit: not sure whether we need to rename it to kube-state-metrics-global.

It has two kube-state-metrics (one Daemonset and one Deployment). If both of them are using the same selector, some wrong pods might be selected. But not sure about it.

Reply:
I spent some time testing a few more scenarios to give a multi-point combination of cases out.

Reply:
Pods will be clearer, I made some combinations, and then came up with some names, it seems that pods will be better, we can make supplementary explanations in the documentation

Reply:
two kube-state-metrics (one Daemonset and one Deployment).
I tested it directly on my side repeatedly and found no duplication.